internal/nix/nixprofile/profile.go

```diff
@@ -204,12 +227,24 @@ func ProfileInstall(args *ProfileInstallArgs) error {
 		fmt.Fprintf(args.Writer, "%s\n", stepMsg)
 	}
 
-	urlForInstall, err := input.URLForInstall()
-	if err != nil {
-		return err
+	var installable string
+	if isInBinaryStore {
+		// TODO savil: change to ContentAddressablePath when that is implemented
+		var err error
+		installable, err = input.PathInBinaryStore()
+		if err != nil {
+			return err
+		}
+	} else if installable == "" {
+		// fallback if PathInBinaryStore is not applicable.
+		var err error
+		installable, err = input.URLForInstall()
+		if err != nil {
+			return err
+		}
```

Comment:
Can we extract line 230 to 244 in it's own separate function that simply returns `(installable, error)`? It is getting a bit hard to read/keep track of